### PR TITLE
[build][tensorflow,pytorch,mxnet][test] upgrade python version

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
- Upgrade python version. Ref: https://python-security.readthedocs.io/vuln/ctypes-buffer-overflow-pycarg_repr.html

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tejaschumbalkar** [comment]
python upgrade on pytorch image blocked due to the patch not available on the conda channel https://anaconda.org/anaconda/python/files

File: test/dlc_tests/sanity/test_safety_check.py [hunk 1]
```diff
@@ -27,11 +27,15 @@
...
+            "py3": ['39611']
         },
         "inference": {
             # for shipping pillow<=6.2.2 - the last available version for py2
-            "py2": ['38449', '38450', '38451', '38452']
+            "py2": ['38449', '38450', '38451', '38452'],
+            # for shipping pyyaml v5.3.1 - blocked on upgrading to v5.4.1 due to dependency on awscli
+            "py3": ['39611']
         },
         "inference-eia": {
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_safety_check.py:40]: Applies to inference-eia as well

**Review by saimidu — COMMENTED** [review]
Please also add a condition for `inference-neuron`.

File: test/dlc_tests/sanity/test_safety_check.py [hunk 2]
```diff
@@ -44,15 +48,20 @@
             "py2": ['36810',
                     # for shipping pillow<=6.2.2 - the last available version for py2
                     '38449', '38450', '38451', '38452'],
+            # for shipping pyyaml v5.3.1 - blocked on upgrading to v5.4.1 due to dependency on awscli
             "py3": ['36810']
```
[the remark this answers is not in the document]
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_safety_check.py]: updated

File: test/dlc_tests/sanity/test_safety_check.py (hunk 1 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_safety_check.py:40]: added

File: test/dlc_tests/sanity/test_safety_check.py [hunk 3]
```diff
@@ -44,15 +54,24 @@
             "py2": ['36810',
                     # for shipping pillow<=6.2.2 - the last available version for py2
                     '38449', '38450', '38451', '38452'],
-            "py3": ['36810']
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_safety_check.py:47]: What is vulnerability 36810?

**Review by saimidu — COMMENTED** [review]
Need to add `"inference-neuron"` as an option in `_get_safety_ignore_list()`

File: test/dlc_tests/sanity/test_safety_check.py (hunk 3 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_safety_check.py:47]: I believe i fixed this, can you refresh your github page

Comment (saimidu) [reply test/dlc_tests/sanity/test_safety_check.py:47]: No, I mean that 36810 has been removed from the list, so just want to make sure we don't encounter it again.

I guess if the test passes, then we will know that it doesn't affect the image anymore.

**Review by saimidu — COMMENTED** [review]
APPROVED.

Let's wait for tests to complete before reverting the safety check unskip.

**Review by saimidu — APPROVED** [review]

**Comment by saimidu** [comment]
Let's merge once the SageMaker test succeeds.

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py [hunk 4]
```diff
@@ -68,6 +68,7 @@ def test_smmodelparallel(sagemaker_session, instance_type, ecr_image, tmpdir, fr
 @pytest.mark.skip_cpu
 @pytest.mark.skip_py2_containers
 @pytest.mark.parametrize("test_script, num_processes", [("smmodelparallel_hvd2_conv_multinode.py", 2)])
+@pytest.mark.skip("Skipping the test due to known issue in TF2.3. Updated binary for SM Model Parallel: https://github.com/aws/deep-learning-containers/pull/837")
```
Comment (tejaschumbalkar) [inline test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py:71]: Related PR: https://github.com/aws/deep-learning-containers/pull/837

**Review by junpuf — APPROVED** [review]

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py (hunk 4 above)
Comment (tejaschumbalkar) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py:71]: and #840

**Pull request merged**